src/core/Core.js

```diff
@@ -415,7 +415,7 @@ class Uppy {
       throttledCalculateProgress(data)
     })
 
-    this.on('core:upload-success', (fileID, uploadResp, uploadURL) => {
```

Comment:
> We could also instead of a getUploadUrl option have a getUploadResponse option, and have that do JSON.parse(xhr.responseText) by default.

Not a bad idea! Could you implement that? I am cautious about merging if we are modifying the callback, making it worse :)